### PR TITLE
GEOMESA-2860 Allow for capability reading v4 Avro written with older …

### PR DESCRIPTION
**Pull request by jnh5y**
…versions of GeoTools

Signed-off-by: Jim Hughes <jnh5y@ccri.com>

### CODE REVIEW

File: geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/serde/Version4CompatTest.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import java.io.{File, FileInputStream}
+
+import org.junit.runner.RunWith
+import org.locationtech.geomesa.features.avro.AvroDataFileReader
+import org.opengis.feature.simple.SimpleFeature
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class Version4CompatTest extends Specification {
```
Comment (elahrvivaz) [inline geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/serde/Version4CompatTest.scala:20]: :100:

File: geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/serde/Version4CompatTest.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  def read(file: File): List[SimpleFeature] = new AvroDataFileReader(new FileInputStream(file)).toList
+
+
+  val files: Seq[String] = Seq("example-v4.avro", "example-v4-with-vis.avro", "example-v5.avro", "example-v5-with-vis.avro")
+  val gt20files: Seq[String] = Seq("example-v4-gt18.avro", "example-v4-with-vis-gt18.avro")
+
+  "Current Reader" should {
+    "read version 4 avro with new Hints package name and version 5 avro" >> {
+       forall(files) { filename =>
+         println(s"Processing $filename")
```
Comment (elahrvivaz) [inline geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/serde/Version4CompatTest.scala:30]: i'll remove the printlns on merge

**Comment by elahrvivaz** [comment]
oh, I guess we also should add the hints check to the v5 serialization... what do you think?

**Comment by elahrvivaz** [comment]
merged as 53d71b90da3d91f70fb6ecf017e83de287bea59b

**Comment by jnh5y** [comment]
> oh, I guess we also should add the hints check to the v5 serialization... what do you think?

Yes.  That'd cover us in the event that we backport writing v5 serialization all the way back to GeoMesa versions which use GT versions <= 20.  *whew*  Good call.

File: geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/serde/Version4CompatTest.scala (hunk 2 above)
Comment (jnh5y) [reply geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/serde/Version4CompatTest.scala:30]: Thank you very much!

**Pull request closed without merging**